src/controller/controller.ts

```diff
@@ -414,6 +426,15 @@ class Controller extends events.EventEmitter {
         return Group.create(groupID);
     }
 
+    /**
+     * Broadcast a network-wide channel change.
+     */
+    public async switchChannel(newChannel: number): Promise<void> {
+        await this.adapter.switchChannel(newChannel);
+
+        this.networkParametersCached = null;// invalidate cache
```

Comment:
Would be better if we could use an event triggered by the adapter to invalidate the cache at the proper time, but it depends on if all stacks (that will implement channel switching) are capable of reporting when the channel actually changed on the coordinator..?